pkg/machines/components/vm/memoryModal.jsx

```diff
@@ -0,0 +1,251 @@
+        };
+        this.close = props.close;
+        this.save = this.save.bind(this);
+        this.onMemoryBlur = this.onMemoryBlur.bind(this);
+        this.onMaxMemoryBlur = this.onMaxMemoryBlur.bind(this);
+        this.onValueChanged = this.onValueChanged.bind(this);
+        this.onUnitsValueChanged = this.onUnitsValueChanged.bind(this);
+        this.dialogErrorSet = this.dialogErrorSet.bind(this);
+    }
+
+    onMemoryBlur(memory) {
+        let maxMemory = convertToUnit(this.state.maxMemory, this.state.maxMemoryUnit, 'MiB');
+
+        memory = convertToUnit(memory, this.state.memoryUnit, 'MiB');
+        memory = clamp(memory, maxMemory, 128); // Until we have method to find out memory requirement of VM's OS, we will use 128MiB
+        memory = convertToUnit(memory, 'MiB', this.state.memoryUnit);
+
+        this.setState({ memory: memory });
+    }
+
+    onMaxMemoryBlur(maxMemory) {
+        const maxNodeMemory = this.props.config.nodeMaxMemory[this.props.vm.connectionName];
+        const memory = convertToUnit(this.state.memory, this.state.memoryUnit, 'MiB');
+
+        maxMemory = convertToUnit(maxMemory, this.state.maxMemoryUnit, 'MiB');
+        maxMemory = clamp(maxMemory, maxNodeMemory, memory);
+        maxMemory = convertToUnit(maxMemory, 'MiB', this.state.maxMemoryUnit);
+
+        this.setState({ maxMemory: maxMemory });
+    }
+
+    onValueChanged(key, value) {
+        const stateDelta = { [key]: value };
+
+        this.setState(stateDelta);
+    }
+
+    onUnitsValueChanged(key, unit) {
+        const memory = Math.round(convertToUnit(this.state[key], this.state[key + "Unit"], unit));
+
+        this.setState({ [key]: memory, [key + "Unit"]: unit });
+    }
+
+    dialogErrorSet(text, detail) {
+        this.setState({ dialogError: text, dialogErrorDetail: detail });
+    }
+
+    save() {
+        const { dispatch, vm } = this.props;
+        const memory = parseInt(convertToUnit(this.state.memory, this.state.memoryUnit, 'KiB'));
+
+        if (vm.memory !== parseInt(convertToUnit(this.state.maxMemory, this.state.maxMemoryUnit, 'MiB'))) {
+            const maxMemory = parseInt(convertToUnit(this.state.maxMemory, this.state.maxMemoryUnit, 'KiB'));
+            dispatch(setMaxMemory(
+                vm,
+                maxMemory
+            ))
+                    .fail((exc) => {
+                        this.dialogErrorSet(_("Could not save maximum memory value:"), exc.message);
+                    })
+                    .then(() => {
+                        if (vm.currentMemory !== parseInt(convertToUnit(this.state.maxMemory, this.state.maxMemoryUnit, 'MiB'))) {
+                            dispatch(setMemory(
+                                vm,
+                                memory
+                            ))
+                                    .fail((exc) => {
+                                        this.dialogErrorSet(_("Could not save memory value:"), exc.message);
```

Comment:
We don't use ':' with dialogErrorSet. Let's keep consistent on how we show error messages. Also we use passive voice, mind switching to that?
For example "VCPU settings could not be saved".
For you shoul db "Maximum memory could not be saved" etc.